cmd/frontend/db/event_logs.go

```diff
@@ -312,6 +360,52 @@ func (l *eventLogs) countPerPeriodBySQL(ctx context.Context, countExpr, interval
 	return counts, nil
 }
 
+func (l *eventLogs) calculatePercentilesPerPeriodBySQL(ctx context.Context, interval, period *sqlf.Query, startDate, endDate time.Time, field string, percentiles []float64, conds []*sqlf.Query) ([]PercentileValue, error) {
+	countByPeriodExprs := []*sqlf.Query{}
+	qExprs := []*sqlf.Query{}
+	for i, p := range percentiles {
+		name := fmt.Sprintf("p%d\n", i)
+		countByPeriodExprs = append(countByPeriodExprs, sqlf.Sprintf("percentile_cont(%d) WITHIN GROUP (ORDER BY (argument->'"+field+"')::integer) AS "+name, p))
```

Comment:
Any reason you didn't use `sqlf.Sprintf` for query interpolation?

```suggestion
		countByPeriodExprs = append(countByPeriodExprs, sqlf.Sprintf("percentile_cont(%d) WITHIN GROUP (ORDER BY (argument->%s)::integer) AS %s", p, field, name))
```

Reply:
I don't think it works with this syntax (the single quotes are required). I can try it again maybe I just mucked something up before.

Reply:
So passing field _does_ work, but the column name does not. I don't think this is possible to quote via sqlf.

However, I think this is ok because we're generating the name directly on the line before and there's no chance for an injection here.

Reply:
Agree, "name" seems good to me.